### PR TITLE
Problem: no integration tests for deploy erc20 with bad denom

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Integration test for https://github.com/crypto-org-chain/gravity-bridge/commit/d984b1562242efab5d7595d4108472ed27eba06f

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Comment by yihuang** [comment]
is this one still relevant? @thomas-nguy

**Comment by thomas-nguy** [comment]
yes it is testing one of the bug fixes from latest gravity version. Will rebase

File: nix/sources.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "gravity-bridge": {
-        "branch": "v2.0.0-cronos",
+        "branch": "thomas/fix-validate-ercdenom",
```
Comment (yihuang) [inline nix/sources.json]: the commit hash don't change?

Comment (thomas-nguy) [reply nix/sources.json]: reverted

**Comment by calvinaco** [comment]
> yes it is testing one of the bug fixes from latest gravity version. Will rebase

Is it possible to point to the bug fix? Trying to understand what "bad denom" means. Thanks!

**Comment by thomas-nguy** [comment]
it tests https://github.com/crypto-org-chain/gravity-bridge/commit/d984b1562242efab5d7595d4108472ed27eba06f

bad denom is a non compliant cosmos denom

File: integration_tests/test_gravity_2.py [hunk 2]
```diff
@@ -367,3 +368,62 @@ def check_gravity_native_tokens():
...
+
+        denom = f"gravity{erc20.address}"
+
+        def check_gravity_native_tokens():
+            "check the balance of gravity native token"
+            return cronos_cli.balance(eth_to_bech32(recipient), denom=denom) == amount
+
+        wait_for_fn("send-to-gravity-native", check_gravity_native_tokens)
+
+        # Deploy a bad cosmos erc20 token
```
Comment (calvinaco) [inline integration_tests/test_gravity_2.py:401]:
```suggestion
        # Deploy a bad cosmos erc20 token with single character
        # Cosmos denom must be 3 ~ 128 characters long and support letters, followed by either
	# a letter, a number or a separator ('/', ':', '.', '_' or '-').
```

Perhaps add comments why the denom is "bad". wdyt?

File: integration_tests/test_gravity_2.py [hunk 3]
```diff
@@ -367,3 +368,64 @@ def check_gravity_native_tokens():
 
         # check that balance is still same
         assert cli.balance(eth_to_bech32(recipient), denom=denom) == max_int
+
+
+def test_gravity_non_cosmos_denom(gravity):
+    if not gravity.cronos.enable_auto_deployment:
```
Comment (yihuang) [inline integration_tests/test_gravity_2.py:374]:
```suggestion
    if gravity.cronos.enable_auto_deployment:
       return
```

**Review by yihuang — APPROVED** [review]

**Pull request merged**